bin/reth/src/ress.rs

```diff
@@ -0,0 +1,60 @@
+use reth_evm::execute::BlockExecutorProvider;
+use reth_network::{protocol::IntoRlpxSubProtocol, NetworkProtocols};
+use reth_network_api::FullNetwork;
+use reth_node_api::BeaconConsensusEngineEvent;
+use reth_node_core::args::RessArgs;
+use reth_primitives::EthPrimitives;
+use reth_provider::providers::{BlockchainProvider, ProviderNodeTypes};
+use reth_ress_protocol::{NodeType, ProtocolEvent, ProtocolState, RessProtocolHandler};
+use reth_ress_provider::{maintain_pending_state, PendingState, RethRessProtocolProvider};
+use reth_tasks::TaskExecutor;
+use reth_tokio_util::EventStream;
+use tokio::sync::mpsc;
+use tracing::info;
+
+/// Install `ress` subprotocol if it's enabled.
+pub fn install_ress_subprotocol<P, E, N>(
+    args: RessArgs,
+    provider: BlockchainProvider<P>,
+    block_executor: E,
+    network: N,
+    task_executor: TaskExecutor,
+    engine_events: EventStream<BeaconConsensusEngineEvent<EthPrimitives>>,
+) -> eyre::Result<Option<mpsc::UnboundedReceiver<ProtocolEvent>>>
+where
+    P: ProviderNodeTypes<Primitives = EthPrimitives>,
+    E: BlockExecutorProvider<Primitives = EthPrimitives> + Clone,
+    N: FullNetwork + NetworkProtocols,
+{
+    if !args.enabled {
+        return Ok(None)
+    }
```

Comment:
can we move this to the callsite so that it's obvious that this is opt-in by looking at main.rs